### PR TITLE
Generate undergraduate test applications

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

Now that we have on our service postgraduate and undergraduate courses I am generating test application for the teacher degree apprenticeship program

## Guidance to review

1. Visit support interface
2. Make sure sidekiq is running
3. Enter tab "Setting" -> "Tasks"
4. Click "Generate test applications"
5. You will see undergraduate applications being created. ✌️

### CODE REVIEW

File: app/workers/generate_test_applications.rb [hunk 1]
```diff
@@ -152,6 +159,30 @@ def courses_from_cycle(year)
...
 
+  def undergraduate_courses_from_cycle(year)
+    recruitment_cycle_year = [RecruitmentCycle.current_year, year].min
+    courses = Course
+      .teacher_degree_apprenticeship
+      .with_course_options
+      .in_cycle(recruitment_cycle_year)
+
+    if courses.count.zero?
+      provider = Provider.all.sample(10).sample || FactoryBot.create(:provider)
```
Comment (tomas-stefano) [inline app/workers/generate_test_applications.rb:170]: I don't care what provider is created those applications and at this point I suspect each developer has your own provider favorite that test locally on QA. Let me know if you have strong opinions about this.

**Comment by tomas-stefano** [comment]
[TDA: Setup the right courses data for Find & Apply](https://trello.com/c/plZ0ECRh/2354-tda-setup-the-right-courses-data-for-find-apply)

File: app/workers/generate_test_applications.rb (hunk 1 above)
Comment (inulty-dfe) [reply app/workers/generate_test_applications.rb:170]: I tend to use provider code 1TZ becuase they have primary and secondary courses by default. Only saying as you asked.

Comment (tomas-stefano) [reply app/workers/generate_test_applications.rb]: @inulty-dfe I pushed one commit using the 1TZ as first try and if there isn't one, they will go to the next or statement and select at random.

**Review by inulty-dfe — APPROVED** [review]

**Pull request merged**